File: 02_ProgrammingFundamentals/_06_ObjectsAndClasses/_02_Exercise/_06_OrderByAge.java
package _02_ProgrammingFundamentals._06_ObjectsAndClasses._02_Exercise;

import java.util.ArrayList;
import java.util.Comparator;
import java.util.List;
import java.util.Scanner;

public class _06_OrderByAge {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        // Read input vale
        String console = scan.nextLine();

        // Add variable
        List<Album> names = new ArrayList<>();


        // Create collection of people
        while (!console.equals("End")) {
            String[] consoleLine = console.split("\\s+");

            String name = consoleLine[0];
            String personalNumber = consoleLine[1];
            int age = Integer.parseInt(consoleLine[2]);

            Album person = new Album(name, personalNumber, age);
            names.add(person);

            console = scan.nextLine();
        }

        // Sort
        names.sort(Comparator.comparing(Album::getAge));

        // Print result
        names.forEach(System.out::println);

        scan.close();
    }
}

// Create practice Album class
class Album{
    private String name ;
    private String personalNUmber;
    private int age;

    public Album(String name, String personalNUmber, int age) {
        this.name = name;
        this.personalNUmber = personalNUmber;
        this.age = age;
    }

    public int getAge() {
        return this.age;
    }

    @Override
    public String toString() {
        return String.format("%s with ID: %s is %d years old.", this.name, this.personalNUmber, this.age);
    }
}

File: 24_C#Fundamentals/_06_ObjectsAndClasses/_02_Exercise/_06_OrderByAge/_06_OrderByAge.cs
using System;

public class _06_OrderByAge
{
    public static void Main(string[] args)
    {
        // Read input vale
        string console = Console.ReadLine();

        // Add variable
        List<Album> names = new List<Album>();


        // Create collection of people
        while (!console.Equals("End"))
        {
            string[] consoleLine = console.Split(" ");

            string name = consoleLine[0];
            string personalNumber = consoleLine[1];
            int age = int.Parse(consoleLine[2]);

            Album person = new Album(name, personalNumber, age);
            names.Add(person);

            console = Console.ReadLine();
        }

        // Print result
        Console.WriteLine(string.Join(Environment.NewLine, names.OrderBy(n => n.Age).Select(e => e.ToString())));
    }
}

// Create practice Album class
class Album
{
    public string Name { get; set; }
    public string PersonalNumber { get; set; }
    public int Age { get; set; }

    public Album(string name, string personalNUmber, int age)
    {
        this.Name = name;
        this.PersonalNumber = personalNUmber;
        this.Age = age;
    }

    public override string ToString()
    {
        return $"{this.Name} with ID: {this.PersonalNumber} is {this.Age} years old.";
    }
}
